### PR TITLE
project completed

### PR DESCRIPTION
**Pull request by rezudox**

### CODE REVIEW

File: doglover/src/components/AllBreeds.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
+import { useEffect, useState } from "react";
+// import { Link } from "react-router-dom";
```
Comment (ngkamperlo) [inline doglover/src/components/AllBreeds.tsx:2]: Is this needed or is it a leftover?

File: doglover/src/components/Breed.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+function Breed() {
+  const {breedId} = useParams()
+  const [images, setImages] = useState([])
+
+  const getSpecificImages = async () => {
+    try {
+      const response = await fetch(`https://dog.ceo/api/breed/${breedId}/images`)
+      const data = await response.json()
+      console.log(data.message)
```
Comment (ngkamperlo) [inline doglover/src/components/Breed.tsx:13]: Is this needed or is it a leftover?

File: doglover/src/components/Breed.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      setImages(data.message)
+      
+    } catch (error:any) {
```
Comment (ngkamperlo) [inline doglover/src/components/Breed.tsx:16]: Could you instead of `any` add another type at this point?

**Review by ngkamperlo — COMMENTED** [review]
Thank you very much for the effort you put into this assignment mate. Can you please answer the following apart from the questions below:
1. Why did you use Typescript without adding any types?
2. Can you please explain why does the application doesn't start with a simple `yarn start`?
3. Why did you add Chakra? Are you familiar with alternative frameworks?
Thanks in advance for all the effort you put in the assignment along with the answers mate!

File: doglover/src/components/AllBreeds.tsx [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+import {Container, Image, Heading, Text, Select, Link} from '@chakra-ui/react';
+
```
Comment (dioannou30) [inline doglover/src/components/AllBreeds.tsx:4]: Hello Alex. Thanks for the effort. When rendering the component there are "some" console errors. Is there a way to resolve them ?

File: doglover/src/components/Breed.tsx [hunk 5]
```diff
@@ -0,0 +1,42 @@
+import {useState, useEffect} from 'react';
+import { useParams } from 'react-router-dom';
+import {Heading, Image, Container,  Grid} from '@chakra-ui/react';
+
```
Comment (dioannou30) [inline doglover/src/components/Breed.tsx:4]: Similarly with AllBreeds.tsx when rendering this component, "some" console errors exist.

**Comment by rezudox** [comment]
Hello,
Thank you for taking the time getting back to me.

1. A mistake on my part. I should have for example : const [images,
setImages] = useState<string[]>([]) and so on.
I'm not really familiar yet with typescript so I did not since I didn't get
any errors despite the fact that I should have. Again my mistake.

2. Because you have to install the dependencies. So for example we have to
run npm install and then npm start.

3. Because I like to work with new things I learn and get some practice
with them. You did not specify if I could use a framework for styling so If
I shouldn't have I apologize.
I believe frameworks as such are easier to maintain especially on large
projects. I am also familiar with material ui.

Yes the commented Link import is indeed a leftover. It is not needed. Same
goes with the console log. It is not needed I should have cleaned up.

To my indeed limited knowledge I think the error only accepts the any or
unknown type. I know the error is a string so there should be a type guard
function for it I think.

Στις Δευ 21 Μαρ 2022 στις 12:34 μ.μ., ο/η Naim Gkamperlo <
***@***.***> έγραψε:

**Comment by rezudox** [comment]
Hello,
Thank you for taking the time to check my task.

Yes, it is happening due to the fact that I did not provide a key for the
render.
As I am not getting a specific id for the breeds rendered, I could fix the
console errors by providing either the breed itself  as a key or the index.
I do believe the breed option as key would be better.
A mistake I did not take care due to my limited time. I know I should have.

Στις Δευ 21 Μαρ 2022 στις 12:47 μ.μ., ο/η Dimitris Ioannou <
***@***.***> έγραψε:

**Pull request closed without merging**